app/gui/src/model/suggestion_database.rs

```diff
@@ -27,6 +28,50 @@ pub use example::Example;
 
 
 
+// ============================
+// === QualifiedNameToIdMap ===
+// ============================
+
+/// A map from fully qualified names (encoded as [`entry::QualifiedNameSegments`]) to
+/// [`entry::Id`]s. The methods of the type provide semantics of a map, while the internal
+/// representation is based on a [`HashMapTree`].
+///
+/// The internal representation conserves memory when storing many paths sharing a common prefix of
+/// segments.
+#[derive(Clone, Debug, Default)]
+struct QualifiedNameToIdMap {
+    tree: ensogl::data::HashMapTree<entry::NameSegment, Option<entry::Id>>,
+}
+
+impl QualifiedNameToIdMap {
+    fn warn_if_exists_and_set(&mut self, path: &entry::QualifiedNameSegments, id: entry::Id) {
+        let value = Some(id);
+        let old_value =
+            self.tree.replace_value_and_traverse_back_pruning_empty_leaf(&path.0, value);
```

Comment:
Please, never use `.0`. Its extremely not readable. What is this `.0`? Use well named variables always.